pheps/phep-0001.md

```diff
@@ -0,0 +1,338 @@
+
+# PHEP Header Preamble
+<a name="phep-header-preamble"></a>
+Each PHEP must begin with an [RFC 2822](https://datatracker.ietf.org/doc/html/rfc2822.html) style header preamble, marked off with a [code fence](https://spec.commonmark.org/current/#fenced-code-blocks). The headers must appear in the following order. Headers marked with "*" are optional. All other headers are required.
+
+      ```
+      PHEP: <phep number>
+      Title: <phep title>
+      Author: <list of authors' real names and optionally, email addrs>
+      Discussions-To: <URL of current canonical discussion thread>
+      Status: <Draft | Accepted | Rejected |
+               Withdrawn | Final | Superseded>
+      Type: <Standards Track | Informational | Process>
+      Content-Type: text/markdown; charset=UTF-8; variant=CommonMark
+    * Requires: <phep numbers>
+      Created: <date created on, in dd-mmm-yyyy format>
+      Post-History: <dates, in dd-mmm-yyyy format,
+                     inline-linked to PHEP discussion threads>
+    * Replaces: <phep number>
+    * Replaced-By: <phep number>
+    * Resolution: <url>
+      ```
+
+The Author header lists the names and optionally the email addresses and/or [ORCIDs](https://orcid.org) of all the authors of the PHEP. The format of the Author header values must be:
+
+`Random J. User <random@example.com> <https://orcid.org/0000-0000-0000-0000>`
+
+if the email address and ORCID are included, and just:
+
+`Random J. User`
+
+for just the name (similarly for ORCID without email address and vice versa).
+
+If there are multiple authors, each should be on a separate line following [RFC 2822](https://datatracker.ietf.org/doc/html/rfc2822.html) continuation line conventions.
+
+The Discussions-To header provides the URL to the current PR discussion thread for the PHEP.
+
+The Type header specifies the [type](#phep-type) of PHEP: Standards Track, Informational, or Process.
+
+The format of a PHEP is specified with a Content-Type header. All PHEPs must use Markdown, and have a value of `text/markdown; charset=UTF-8; variant=CommonMark`.
+
+The Created header records the date that the PHEP was assigned a number, while Post-History is used to record the dates of pushes to the pull request for the PHEP. Both sets of dates should be in `dd-mmm-yyyy` format, e.g. `14-Aug-2001`.
+
+PHEPs may have a Requires header, indicating the PHEP numbers that this PHEP depends on.
+
+PHEPs may also have a Replaced-By header indicating that a PHEP has been rendered obsolete by a later document; the value is the number of the PHEP that replaces the current document. The newer PHEP must have a Replaces header containing the number of the PHEP that it rendered obsolete.
+
+# Auxiliary Files
+<a name="auxiliary-files"></a>
+PHEPs may include auxiliary files such as diagrams. Such files must be placed in a subdirectory called `phep-XXXX`, where "XXXX" is the PHEP number. There are no constraints on the names used in files.
```

Comment:
PEP 1 and previous drafts provide an option of having the auxiliary files in the main directory with a convoluted naming scheme. It seems simple to just have one way to do it.

Reply:
Agreed, let's just have one way to do it.